supabase/functions/oauth/encrypt-config.ts

```diff
@@ -48,19 +50,23 @@ export async function encryptConfig(req: Record<string, any>) {
   });
 
   let responseData = JSON.stringify(await response.json());
-  console.log(responseData);
 
   // If we can find client_id or client_secret in plaintext in the response,
   // it's not secure to return this response!
   if (
-    responseData.includes(oauth2_client_id) ||
-    responseData.includes(oauth2_client_secret)
+    oauth2_client_id != null &&
+    oauth2_client_secret != null &&
+    (responseData.includes(oauth2_client_id) ||
+      responseData.includes(oauth2_client_secret))
   ) {
     return new Response(
       JSON.stringify({
-        error: "exposed_secret",
-        description: `client_id and client_secret were not encrypted as part of this request.
+        error: {
```

Comment:
It seems like this response ought to have a 400 (or maybe even a 500?) status, rather than the status from the config-encryption response.

If we hit this condition, then it's most likely not due to anything the user did or has control over. It's our problem because the schema is incorrect. It's of course arguable that it's the users' problem because they maybe could be using a connector that they built, so :shrug: